src/jobs/BenchmarkJob.jl

```diff
@@ -70,6 +70,17 @@ function BenchmarkJob(submission::JobSubmission)
             error("invalid argument to `vs` keyword")
         end
         against = againstbuild
+    elseif submission.prnumber !== nothing
+        # if there is a PR number, we compare against the base branch
+        repo = joinpath(workdir, "julia")
+        if isdir(joinpath(repo, ".git"))
+            gitreset!(repo)
+        else
+            gitclone!(submission.config.trackrepo, repo)
+        end
+        run(`$(git()) -C $repo fetch --quiet origin +refs/pull/$(submission.prnumber)/head:`)
```

Comment:
it feels very weird to do git operations here, since our `build_julia!` step knows so much more about users, tempdirs, and mirrors

Reply:
Maybe not quite as convenient as getting merge-base directly, but it does have the ability to list all commits present on one branch but not on another (`origin/master...refs/pull/#/head`)

Reply:
So you propose using the parent of the first commit in that selection? Wouldn't that "break" when there's a merge commit with master in the PR?

Reply:
I don't think the first commit can be a merge. But anyways, GitHub API appears to have merge_base_commit for this:
https://docs.github.com/en/rest/commits/commits#compare-two-commits

Reply:
https://github.com/JuliaWeb/GitHub.jl/pull/202